flocker/provision/_install.py

```diff
@@ -34,21 +38,50 @@
 
 ARCHIVE_BUCKET = 'clusterhq-archive'
 
-CLUSTERHQ_REPO = {
-    'fedora-20': "https://s3.amazonaws.com/{archive_bucket}/"
-                 "fedora/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
-                     archive_bucket=ARCHIVE_BUCKET,
-                 ),
-    'centos-7': "https://s3.amazonaws.com/{archive_bucket}/"
-                "centos/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
-                    archive_bucket=ARCHIVE_BUCKET,
-                    ),
-    # FLOC-1828 TODO - use ubuntu rather than ubuntu-testing
-    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
-                'ubuntu-testing/14.04/$(ARCH)'.format(
-                    archive_bucket=ARCHIVE_BUCKET
-                    ),
-}
+
+def get_repository_url(distribution, flocker_version):
+    """
+    Return the URL for the repository of a given distribution.
+
+    Note that for ``yum``-using distributions this gives the URL to a package.
```

Comment:
...to a package that adds entries to /etc/yum.repos.d

For ``apt``-using distributions, this gives the URL for a repo containing a Packages(.gz) file.